scipy/sparse/compressed.py

```diff
@@ -399,42 +399,83 @@ def multiply(self, other):
             elif self.shape == (1,1):
                 return other._mul_scalar(self.toarray()[0, 0])
             # A row times a column.
-            elif self.shape[1] == other.shape[0] and self.shape[1] == 1:
+            elif self.shape[1] == 1 and other.shape[0] == 1:
                 return self._mul_sparse_matrix(other.tocsc())
-            elif self.shape[0] == other.shape[1] and self.shape[0] == 1:
+            elif self.shape[0] == 1 and other.shape[1] == 1:
                 return other._mul_sparse_matrix(self.tocsc())
             # Row vector times matrix. other is a row.
             elif other.shape[0] == 1 and self.shape[1] == other.shape[1]:
                 other = dia_matrix((other.toarray().ravel(), [0]),
-                                    shape=(other.shape[1], other.shape[1]))
+                                   shape=(other.shape[1], other.shape[1]))
                 return self._mul_sparse_matrix(other)
             # self is a row.
             elif self.shape[0] == 1 and self.shape[1] == other.shape[1]:
                 copy = dia_matrix((self.toarray().ravel(), [0]),
-                                    shape=(self.shape[1], self.shape[1]))
+                                  shape=(self.shape[1], self.shape[1]))
                 return other._mul_sparse_matrix(copy)
             # Column vector times matrix. other is a column.
             elif other.shape[1] == 1 and self.shape[0] == other.shape[0]:
                 other = dia_matrix((other.toarray().ravel(), [0]),
-                                    shape=(other.shape[0], other.shape[0]))
+                                   shape=(other.shape[0], other.shape[0]))
                 return other._mul_sparse_matrix(self)
             # self is a column.
             elif self.shape[1] == 1 and self.shape[0] == other.shape[0]:
                 copy = dia_matrix((self.toarray().ravel(), [0]),
-                                    shape=(self.shape[0], self.shape[0]))
+                                  shape=(self.shape[0], self.shape[0]))
                 return copy._mul_sparse_matrix(other)
             else:
                 raise ValueError("inconsistent shapes")
-        # Dense matrix.
-        if isdense(other):
-            if self.shape == other.shape:
-                ret = self.tocoo()
-                ret.data = np.multiply(ret.data, other[ret.row, ret.col]
-                                       ).view(np.ndarray).ravel()
-                return ret
+
+        other = np.atleast_2d(other)
```

Comment:
This doesn't seem necessary and may not be quite right. If you get other objects like sparse arrays (but not captured by an if-clause above) that don't react well to ``atleast_2d``, you may break code that used to work fine with ``return np.multiply(self.todense(), other)``.